packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/ui/viewModel/ConnectionStateViewModel.kt

```diff
@@ -87,16 +89,23 @@ class ConnectionStateViewModel(
     private val coroutineScope: CoroutineScope
 ) : DataSourceManager.Listener, JdbcDriverManager.Listener {
     internal var connectionSaga = ConnectionSaga(project, ::emitSelectedConnectionChanged)
-    val connectionState = MutableStateFlow(ConnectionState.default())
+    internal val mutableConnectionState = MutableStateFlow(ConnectionState.default())
+    val connectionState: StateFlow<ConnectionState> = mutableConnectionState
```

Comment:
```suggestion
    val connectionState = mutableConnectionState.asStateFlow()
```

It's a bit shorter and we don't need to specify the type twice.

Reply:
Addressed this in my [recent commit](https://github.com/mongodb/intellij/pull/170/commits/cf48496cdfb1688172a935bf7141d80b44ce7efc).